test/test_filepost.py

```diff
@@ -1,15 +1,49 @@
-import unittest
-
-from urllib3.filepost import encode_multipart_formdata, iter_fields
+import psutil
+import io
+import sys
+
+if sys.version_info >= (2, 7):
+    import unittest
+else:
+    import unittest2 as unittest
+
+from urllib3.filepost import (
+    encode_multipart_formdata,
+    iter_fields,
+    MultipartEncoderGenerator
+)
 from urllib3.fields import RequestField
 from urllib3.packages.six import b, u
 
 
 BOUNDARY = '!! test boundary !!'
 
 
-class TestIterfields(unittest.TestCase):
+class TestMultipartEncoderGenerator(unittest.TestCase):
+    def test_len(self):
+        fieldsets = [
+            [('k', 'v'), ('k2', 'v2')],
+            [('k', b'v'), (u('k2'), b'v2')],
+            [('k', b'v'), (u('k2'), 'v2')],
+            [('foo', b'a'), ('foo', b'b')],
+        ]
+
+        for fields in fieldsets:
+            encoded, _ = encode_multipart_formdata(fields, boundary=BOUNDARY)
+            encoded = b''.join(encoded)
+            self.assertEqual(len(encoded), len(MultipartEncoderGenerator(fields, boundary=BOUNDARY)))
+
+    def test_memory_usage(self):
+        MEGABYTE = 1024 * 1024
+        data = b'x' * MEGABYTE  # 1 MB of data
+        start_memory = psutil.Process().memory_info().rss
+
+        encoded, _ = encode_multipart_formdata([('k', data)], boundary=BOUNDARY)
+        end_memory = psutil.Process().memory_info().rss
+        self.assertTrue(end_memory <= start_memory + MEGABYTE)
```

Comment:
I think I'll just drop this test, maybe I can come up with a better way to test that the data is not copied or duplicated.